packages/Laravel/src/Rector/Class_/InlineValidationRulesToArrayDefinitionRector.php

```diff
@@ -0,0 +1,184 @@
+    {
+        return [Node\Stmt\Class_::class];
+    }
+
+    /**
+     * @param Node\Stmt\Class_ $node
+     * @return Node|null
+     */
+    public function refactor(Node $node): ?Node
+    {
+        $className = $this->getName($node);
+        if ($className === null || !is_subclass_of($className, 'Illuminate\Foundation\Http\FormRequest', true)) {
+            return null;
+        }
+
+        $rules = $node->getMethod('rules');
+        if ($rules === null || $rules->stmts === null) {
+            return null;
+        }
+
+        foreach ($rules->stmts as $statement) {
+            if (!$statement instanceof Node\Stmt\Return_) {
+                continue;
+            }
+
+            $expression = $statement->expr;
+            if (!$expression instanceof Node\Expr\Array_) {
+                continue;
+            }
+
+            foreach ($expression->items as $item) {
+                if (!$item->value instanceof Node\Scalar\String_ && !$item->value instanceof Node\Expr\BinaryOp\Concat) {
+                    continue;
+                }
+
+                $newRules = $this->transformRulesSet($item->value);
+                foreach ($newRules as &$newRule) {
+                    if (!($newRule instanceof Node\Expr\BinaryOp\Concat)) {
+                        continue;
+                    }
+
+                    try {
+                        $fullString = $this->transformConcatExpression($newRule);
+                    } catch (\LogicException $exception) {
+                        continue;
+                    }
+
+                    $matchesExist = preg_match('/^exists:(\w+),(\w+)$/', $fullString, $matches);
+                    if ($matchesExist === false || $matchesExist === 0) {
+                        continue;
+                    }
+
+                    $newRule = new Node\Expr\StaticCall(
+                        new Node\Name('\Illuminate\Validation\Rule'),
+                        'exists',
+                        [
+                            new Node\Arg(
+                                new Node\Expr\ClassConstFetch(
+                                    new Node\Name($matches[1]),
+                                    'class'
+                                )
+                            ),
+                            new Node\Arg(new Node\Scalar\String_($matches[2])),
+                        ]
+                    );
+                }
+
+                $item->value = new Node\Expr\Array_(
+                    array_map(function (Node\Expr $rule): Node\Expr\ArrayItem {
+                        return new Node\Expr\ArrayItem($rule);
+                    }, $newRules)
+                );
+            }
+        }
+
+        return $node;
+    }
+
+    /**
+     * @param Node\Expr $expr
+     * @return Node\Expr[]
+     */
+    private function transformRulesSet(Node\Expr $expr): array
+    {
+        if ($expr instanceof Node\Scalar\String_) {
+            $parts = preg_split('/\|/', $expr->value);
```

Comment:
Better `explode`, since there can be just 1 rule as well, right?

Reply:
I forgot about `explode` function :)